addons/website_sale/controllers/main.py

```diff
@@ -887,6 +887,12 @@ def _get_shop_payment_values(self, order, **kwargs):
             return_url= '/shop/payment/validate',
             bootstrap_formatting= True
         )
+        if order.partner_id.company_id and order.partner_id.company_id.id != order.company_id.id:
```

Comment:
This might be too restrictive if we were to someday relax the security rules on payment providers.
But I guess it's unlikely to happen and I am ok with it as is right now.